orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfiguration.groovy

```diff
@@ -101,18 +105,32 @@ class CloudDriverConfiguration {
           .create(type)
     }
 
-    public <T> T buildReadOnlyService(Class<T> type) {
-      if (cloudDriverConfigurationProperties.cloudDriverReadOnlyBaseUrl == cloudDriverConfigurationProperties.cloudDriverBaseUrl) {
+    @CompileDynamic
+    private <T> SelectableService buildReadOnlyService(Class<T> type) {
+      if (cloudDriverConfigurationProperties.cloudDriverReadOnlyBaseUrls*.baseUrl == [cloudDriverConfigurationProperties.cloudDriverBaseUrl]) {
         log.info("readonly URL not configured for clouddriver, using writeable clouddriver $cloudDriverConfigurationProperties.cloudDriverBaseUrl for $type.simpleName")
       }
 
-      return buildService(type, cloudDriverConfigurationProperties.cloudDriverReadOnlyBaseUrl)
+      return new SelectableService(
+        serviceSelectors: cloudDriverConfigurationProperties.cloudDriverReadOnlyBaseUrls.collect {
+          def selector = new DefaultServiceSelector(buildService(type, it.baseUrl), it.config)
+
+          def selectorClass = it.config?.selectorClass as Class<ServiceSelector>
```

Comment:
worth having a list of selectors here that all have to match? If I wanted to specifically route the orchestrations from a particular app somewhere that would allow it